packages/expo-notifications/android/src/main/java/expo/modules/notifications/notifications/triggers/NotificationTriggers.kt

```diff
@@ -0,0 +1,183 @@
+package expo.modules.notifications.notifications.triggers
+
+import android.os.Parcel
+import expo.modules.notifications.notifications.interfaces.NotificationTrigger
+import expo.modules.notifications.notifications.interfaces.SchedulableNotificationTrigger
+import kotlinx.parcelize.IgnoredOnParcel
+import kotlinx.parcelize.Parceler
+import kotlinx.parcelize.Parcelize
+import java.io.Serializable
+import java.util.Calendar
+import java.util.Date
+
+@Parcelize
+open class ChannelAwareTrigger(open val channelId: String?) :
+  NotificationTrigger, Serializable {
+
+  override fun describeContents(): Int = 0
+
+  override fun getNotificationChannel(): String? {
+    return channelId
+  }
+}
+
+/**
+ * A schedulable trigger representing a notification to be scheduled once per day.
+ */
+@Parcelize
+class DailyTrigger(override val channelId: String?, val hour: Int, val minute: Int) : ChannelAwareTrigger(channelId), SchedulableNotificationTrigger {
+
+  override fun nextTriggerDate(): Date? {
+    val nextTriggerDate = Calendar.getInstance()
+    nextTriggerDate[Calendar.HOUR_OF_DAY] = hour
+    nextTriggerDate[Calendar.MINUTE] = minute
+    nextTriggerDate[Calendar.SECOND] = 0
+    nextTriggerDate[Calendar.MILLISECOND] = 0
+    val rightNow = Calendar.getInstance()
+    if (nextTriggerDate.before(rightNow)) {
+      nextTriggerDate.add(Calendar.DATE, 1)
+    }
+    return nextTriggerDate.time
+  }
+}
+
+/**
+ * A schedulable trigger representing notification to be scheduled only once at a given moment of time.
+ */
+@Parcelize
+class DateTrigger(override val channelId: String?, timestamp: Long) : ChannelAwareTrigger(channelId), SchedulableNotificationTrigger {
+  val triggerDate = Date(timestamp)
```

Comment:
AFAICT `triggerDate` doesn't need to be an instance variable?


This declaration could be moved to `nextTriggerDate` and we could remove the `companion object` entirely (and the private constructor)?

Reply:
Moving this would cause a change to the public methods of the class, and require changes in various other files